docs/pages/usage-spark.md

```diff
@@ -1,3 +1,102 @@
 ---
 title: Spark
----
\ No newline at end of file
+---
+
+Almond comes with a Spark integration module called *almond-spark*, which allows you to connect to a Spark cluster and
+to run Spark calculation interactively from a Jupyter notebook.
+
+It is based on [ammonite-spark](https://github.com/alexarchambault/ammonite-spark), adding Jupyter specific features
+such as progress bars and cancellation for running Spark computations.
+
+*ammonite-spark* handles loading Spark in a clever way, and does not rely on a specific Spark distribution.
+Because of that, you can use it with any Spark 2.x version.
+The only limitation is that the Scala version of Spark and the running Almond kernel must match, so make sure your
+kernel uses the same Scala version as your Spark cluster.
+Spark 2.0.x - 2.3.x requires Scala 2.11. Spark 2.4.x supports both Scala 2.11 and 2.12.
+
+For more information, see the [README](https://github.com/alexarchambault/ammonite-spark/blob/master/README.md) of ammonite-spark.
+
+To use it, import the *almond-spark* dependency as well as Spark 2.x itself.
+
+```scala
+import $ivy.`org.apache.spark::spark-sql:2.4.0` // Or use any other 2.x version here
+import $ivy.`sh.almond::almond-spark:0.3.0`
+```
+
+Usually you want to disable logging in order to avoid polluting your cell outputs:
+
+```scala
+import org.apache.log4j.{Level, Logger}
+Logger.getLogger("org").setLevel(Level.OFF)
+
+```
+
+Then create a `SparkSession` using the `NotebookSparkSessionBuilder` provided by *almond-spark*:
+
+```scala
+import org.apache.spark.sql._
+
+val spark = {
+  NotebookSparkSession.builder()
+    .master("local[*]")
+    .getOrCreate()
+}
+```
+
+When running this, you should see that the cell output contains a link to the Spark UI.
+
+Note the use of `NotebookSparkSession.builder()`, instead of `SparkSession.builder()` that one would use when e.g. writing a Spark job.
+
+The builder returned by `NotebookSparkSession.builder()` extends the one of `SparkSession.builder()`,
+so that one can call `.appName("foo")`, `.config("key", "value")`, etc. on it.
+
+Now you can get a `SparkContext` from the `SparkSession` and run Spark calculations.
+
+```scala
+def sc = spark.sparkContext
+
+val rdd = sc.parallelize(1 to 100000000, 100)
+
+val n = rdd.map(_ + 1).sum()
+```
+
+When you execute a Spark `Action` like run you should see a progress bar, showing the progress of the running Spark job,
+as well as a link to cancel the job.
```

Comment:
For now, the kill button needs some Javascript code to run from the front-end, to send a comm message to the kernel. So it only works from the Jupyter classic UI (jupyterlab doesn't run Javascript out-of-the-box).

I guess it shouldn't be shown by default (and only some Javascript code would make it visible, ensuring some Javascript code can actually be run).

```suggestion
as well as a link to cancel the job if you are using the Jupyter classic UI.
```